### DISCUSSION TITLE
RG350_auto_ra_installer config files vs Adam ones

### DISCUSSION THREAD

**Discussion by m45t3r** [eduardofilo/RG350_adam_image#181] [RetroArch topics]
Here is something that I realized after flashing the 1.4 release: my RetroArch video output config went back to fit the whole screen for most emulators (the ones I tested: SNES, GB/GBC, GBA), without maintaining the correct aspect ratio. However before it was keeping the correct aspect ratio, and in the SNES case it was even using the integer scale that I prefer since it is **much** sharper (at the expense of losing some screen state). I don't remember of manually configuring it, but them I remember that when I upgrade the RetroArch cores on 1.3.1 with [RG350_auto_ra_installer](https://github.com/eduardofilo/RG350_auto_ra_installer) I would always mark both options (`Uninstall previous wrappers` and `Install config`).

So I tried to install RG350_auto_ra_installer's [v2022-02-03](https://github.com/eduardofilo/RG350_auto_ra_installer/releases/tag/v2022-02-03) and there I have the emulators with correct aspect ratio and integer scaling in the cases where the screen is not absurdly small.

Now, this is not necessary an issue because I realize that everyone has different preferences in relation of video output. So I mostly opened this issue to leave this workaround for those people that prefer correct aspect ratio without messing each individual emulator config.

BTW (and this is actually an issue), another thing I noted is that it is impossible to overwrite the by using `Main Menu (press B on Quick Menu) -> Configuration File -> Save Current Configuration` in the "vanilla" Adam firmware (this is also what makes the incorrect aspect ratio issue more annoying, since it is not possible to fix this unless you delete the original configuration for the core). After flashing `RG350_auto_ra_installer`, the issue is gone, so this is probably some issue of Unix permission.

Edit: installing RG350_auto_ra_installer config unfortunately also causes the Analogue to not be mapped to Directional anymore on systems that don't support Analogue. However this is not much of an issue for me (I find it kinda strange playing a e.g. SNES game with Analogue, but some people probably like this option).

**Comment by eduardofilo** [comment]
Complex question.
The settings between the image and the auto_ra_installer have diverged since some time. For this reason, in the installation instructions of the auto_ra_installer to be used on Adam, I indicate that the second check should not be marked.
I also like pixel perfect in general, but people asked to expand to full screen mainly to use the image in RG280.
About the problems you say to reset those settings, it is not an issue. You have to watch carefully the files that come into play. Review this FAQs:

https://github.com/eduardofilo/RG350_adam_image/wiki/En:-7.-FAQ#q3-why-are-the-override-settings-in-retroarch-not-preserved

https://github.com/eduardofilo/RG350_adam_image/wiki/En:-7.-FAQ#q6-i-cant-modify-retroarch-config-error-failed-saving-config-to-is-displayed

**Comment by m45t3r** [comment]
> About the problems you say to reset those settings, it is not an issue. You have to watch carefully the files that come into play. Review this FAQs

Huh... Strange that after reinstalling the config from RG350_auto_ra_installer, the config saves correctly even without unloading the content.

**Reply by eduardofilo** [reply]
The configuration in auto_ra_installer is different and therefore the cascade of settings that exists at the different levels (general, core, content, ROM) can interfere in different ways. I have tried to make changes on a clean installation of the image and I have not had a problem. It is best to review the files from terminal (by SSH) while making changes and check for differences with `diff` or grepping as shown in FAQ number 3.

If you want, tell me what specific change you wanted to make in the general configuration and I will analyze it.
